File: tagisClient/src/app/inventory/order.ts
import { Product } from './product';

export class Order {
  _oid: number;
  order_status: string = '';
  created_date: string = '';
  last_modified_date: string = '';
  products: Array<Product>;
  subtotal: number;
  taxes: number;
  total: number;
  company: Object;
  company_title: string = '';
  logo: string; // Company logo
  customer_name: string = '';
  customer_email: string = '';
  customer_phone: string = '';
  customer_address1: string = '';
  customer_address2: string = '';
  customer_city: string = '';
  customer_state: string = '';
  customer_zip: string = '';
  shipping_name: string = '';
  shipping_address1: string = '';
  shipping_address2: string = '';
  shipping_city: string = '';
  shipping_state: string = '';
  shipping_zip: string = '';
  order_source: string = '';
  ref: string = '';
  shipping_number: string = '';
  shipping_carrier: string = '';
  send_order_receipt: Boolean;
  receipt_email_sent: Date;
  send_shipping_notification: Boolean;
  shipping_email_sent: Date;
}

File: tagisApi/tagisApi/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace tagisApi.Models
{
    [Table("Orders")]
    public class Order
    {
        enum OrderStatus
        {
            New,
            Canceled,
            Shipped
        }

        [Key]
        [Required]
        public int _oid { get; set; }

        public DateTime CreatedDate { get; set; }

        public DateTime LastModifiedDate { get; set; }

        [Required]
        public IList<OrderItem> Products { get; set; }

        [Required]
        public double Taxes { get; set; }

        public double ShippingFees { get; set; }

        [Required]
        public double Subtotal { get; set; }

        [Required]
        public double Total { get; set; }

        [Required]
        private int CompanyId { get; set; }

        public string OrderSource { get; set; }

        [Required]
        private int CustomerId { get; set; }

        private string ShippingAddress1 { get; set; }

        private string ShippingAddress2 { get; set; }

        private string ShippingCity { get; set; }

        private string ShippingState { get; set; }

        private string ShippingZip { get; set; }

        public string RefNumber { get; set; }

        public string TrackingNumber { get; set; }

        public string ShippingCarrier { get; set; }

        public DateTime ReceiptEmailSent { get; set; }

        public DateTime ShippingEmailSent { get; set; }

        public DateTime OrderNotificationEmail { get; set; }
    }
}